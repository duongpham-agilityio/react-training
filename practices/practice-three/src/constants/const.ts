export const LIMIT_QUANTITY: number = 5;
