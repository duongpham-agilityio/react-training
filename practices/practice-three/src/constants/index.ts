export * from './url';
export * from './message';
export * from './variable';
export * from './title';
export * from './regex';
export * from './time';
export * from './status';
