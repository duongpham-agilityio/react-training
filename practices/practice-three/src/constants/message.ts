interface IMessages {
  EMPTY: string;
  FAIL_TO_FETCH: string;
  ADD_TO_CART_SUCCESS: string;
  ADD_TO_CART_FAIL: string;
  REMOVE_FORM_CART_SUCCESS: string;
  REMOVE_FORM_CART_FAIL: string;
}

export const MESSAGES: Readonly<IMessages> = {
  EMPTY: 'No data',
  FAIL_TO_FETCH: 'Something went wrong!! Please reload page.',
  ADD_TO_CART_SUCCESS: 'Add to cart successful!',
  ADD_TO_CART_FAIL: 'Add to cart fail!',
  REMOVE_FORM_CART_SUCCESS: 'Remove from cart successful!',
  REMOVE_FORM_CART_FAIL: 'Remove from cart fail!',
};
