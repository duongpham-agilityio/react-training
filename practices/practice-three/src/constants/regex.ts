export const REGEXPS = {
  EMAIL: /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/,
};
