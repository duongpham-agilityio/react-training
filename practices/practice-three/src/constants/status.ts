export const STATUS = {
  NOT_FOUND: 404,
  UNAUTHORIZED: 401,
  SUCCESS: 200,
};
