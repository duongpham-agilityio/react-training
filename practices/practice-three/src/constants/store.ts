export const STORES = {
  CART: 'cart',
  FAVORITE: 'favorites',
  AUTHENTICATION: 'auth',
};
