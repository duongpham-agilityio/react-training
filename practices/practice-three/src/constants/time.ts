export const TIMES = {
  DEBOUNCE: 500,
  TOAST: 1000,
};
