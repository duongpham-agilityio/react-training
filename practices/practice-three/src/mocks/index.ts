export * from './productCard';
export * from './filterBar';
export * from './inputForm';
export * from './pagination';
