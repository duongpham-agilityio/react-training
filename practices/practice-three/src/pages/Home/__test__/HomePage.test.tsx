import { act, fireEvent, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import {
  QueryClient,
  QueryClientProvider,
  useQuery,
} from '@tanstack/react-query';

// Components
import HomePage from '..';

// Mocks
import { products } from '@/mocks';

// Constants
import { MESSAGES } from '@/constants';

const queryClient: QueryClient = new QueryClient();

jest.mock('@tanstack/react-query', () => ({
  ...jest.requireActual('@tanstack/react-query'),
  useQuery: jest.fn(),
}));

const setup = () =>
  render(
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>
    </QueryClientProvider>,
  );

describe('Home page', () => {
  it('Match to snapshot', () => {
    (useQuery as jest.Mock).mockReturnValue({
      isLoading: false,
      data: products,
    });
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('With Loading', () => {
    (useQuery as jest.Mock).mockReturnValue({
      isLoading: true,
    });
    const { getByText } = setup();

    expect(getByText('Loading...')).toBeDefined();
  });

  it('With Error', () => {
    (useQuery as jest.Mock).mockReturnValue({
      isError: true,
    });
    const { getByText } = setup();

    expect(getByText(MESSAGES.FAIL_TO_FETCH)).toBeDefined();
  });

  it('Change filter', async () => {
    (useQuery as jest.Mock).mockReturnValue({
      isLoading: false,
      data: products,
    });
    const { container, getByPlaceholderText } = setup();

    expect(container).toMatchSnapshot();

    const input: HTMLInputElement = getByPlaceholderText(
      'Search',
    ) as HTMLInputElement;

    act(() => {
      fireEvent.change(input, {
        target: {
          value: 'abcd',
        },
      });
    });

    expect(input.value).toBe('abcd');
  });

  it('Add to cart', async () => {
    (useQuery as jest.Mock).mockReturnValue({
      isLoading: false,
      data: products,
    });
    const { container } = setup();
    const buttons = container.querySelectorAll(
      'button[aria-label="Button add to cart"]',
    );
    const button = buttons[0];
    const mock = jest.fn();

    button.addEventListener('click', mock);

    fireEvent.click(button);

    expect(mock).toBeCalled();
  });

  it('Add to wishlist', async () => {
    (useQuery as jest.Mock).mockReturnValue({
      isLoading: false,
      data: products,
    });
    const { container } = setup();
    const buttons = container.querySelectorAll(
      'button[aria-label="Button add to favorite"]',
    );
    const button = buttons[0];
    const mock = jest.fn();

    button.addEventListener('click', mock);

    fireEvent.click(button);

    expect(mock).toBeCalled();
  });
});
