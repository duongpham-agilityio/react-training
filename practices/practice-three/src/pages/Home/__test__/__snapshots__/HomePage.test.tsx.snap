// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Change filter 1`] = `
<div>
  <div
    class="chakra-stack css-1ey9w6j"
  >
    <div
      class="chakra-input__group css-viwvyh"
      data-group="true"
    >
      <input
        class="chakra-input css-190aajt"
        placeholder="Search"
        value=""
      />
      <div
        class="css-1q3ec5g"
      >
        <svg
          fill="none"
          height="30"
          width="42"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M25.507 7.2H1.8a1.8 1.8 0 0 1 0-3.6h23.707a5.402 5.402 0 0 1 10.186 0H40.2a1.8 1.8 0 0 1 0 3.6h-4.507a5.402 5.402 0 0 1-10.186 0ZM40.2 26.4a1.8 1.8 0 0 0 0-3.6H16.493a5.402 5.402 0 0 0-10.186 0H1.8a1.8 1.8 0 0 0 0 3.6h4.507a5.402 5.402 0 0 0 10.186 0H40.2Z"
            fill="#6A6A6A"
          />
        </svg>
      </div>
    </div>
    <ul
      class="css-1c8ne0r"
      role="list"
    >
      <li
        class="css-0"
      >
        <a
          class="chakra-button css-16s5d4s"
          href="/?category=all"
        >
          All
        </a>
      </li>
      <li
        class="css-0"
      >
        <a
          class="chakra-button css-16s5d4s"
          href="/?category=fashion"
        >
          Fashion
        </a>
      </li>
      <li
        class="css-0"
      >
        <a
          class="chakra-button css-16s5d4s"
          href="/?category=electrics"
        >
          Electrics
        </a>
      </li>
      <li
        class="css-0"
      >
        <a
          class="chakra-button css-16s5d4s"
          href="/?category=books"
        >
          Books
        </a>
      </li>
    </ul>
  </div>
  <h2
    class="chakra-heading css-p4222r"
  >
    Trending Items
  </h2>
  <div
    class="css-3hictu"
  >
    <div
      class="css-1gvp5h0"
    >
      <a
        class="chakra-card css-1wt50ai"
        href="/detail/1"
      >
        <div
          class="chakra-stack css-m4lmkl"
        >
          <div
            class="css-1ql0rgo"
          >
            <img
              alt="Tasty Plastic Mouse"
              class="chakra-image css-13rbvqk"
              src="https://loremflickr.com/640/480/fashion"
            />
          </div>
          <div
            class="chakra-card__body css-dspdqf"
          >
            <div
              class="chakra-stack css-1luc48q"
            >
              <div
                class="css-13rbvqk"
              >
                <h2
                  class="chakra-heading css-n93nk6"
                >
                  Tasty Plastic Mouse
                </h2>
                <p
                  class="chakra-text css-1286dop"
                >
                  Quos tempora adipisci quidem ipsum.
                </p>
                <span
                  class="chakra-badge css-1dm3gsd"
                >
                  <div
                    class="css-artlef"
                  >
                    available
                  </div>
                </span>
              </div>
              <div
                class="chakra-stack css-bot922"
              >
                <p
                  class="chakra-text css-17nnq38"
                >
                  $
                  430
                </p>
                <div
                  class="chakra-stack css-1y3f6ad"
                >
                  <button
                    aria-label="Button add to cart"
                    class="chakra-button css-p7mkev"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="27"
                      width="28"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M25.502 3.562H5.821L4.139.548A1.069 1.069 0 0 0 3.206 0H1.069a1.069 1.069 0 1 0 0 2.137h1.51l1.659 2.975 3.827 8.469.004.008.337.746-3.842 4.099a1.069 1.069 0 0 0 .658 1.793l3.503.4c4.38.5 8.804.5 13.186 0l3.502-.4a1.069 1.069 0 1 0-.242-2.124l-3.503.4c-4.22.483-8.48.483-12.7 0l-1.39-.159 2.82-3.008c.027-.03.052-.06.076-.09l1.072.14a21.81 21.81 0 0 0 4.534.114 13.237 13.237 0 0 0 9.071-4.658l.824-.976c.028-.032.053-.067.077-.103l1.535-2.34c1.088-1.657-.101-3.86-2.085-3.86Zm-14.949 9.539a.69.69 0 0 1-.538-.397l-.002-.004-3.164-7h18.653c.284 0 .454.314.298.551l-1.5 2.286-.782.926a11.1 11.1 0 0 1-7.607 3.906 19.674 19.674 0 0 1-4.089-.103l-1.269-.165Z"
                        fill="#979797"
                        fill-rule="evenodd"
                      />
                      <path
                        d="M7.48 22.442a2.137 2.137 0 1 0 0 4.275 2.137 2.137 0 0 0 0-4.275ZM21.017 24.58a2.137 2.137 0 1 1 4.275 0 2.137 2.137 0 0 1-4.275 0Z"
                        fill="#979797"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Button add to favorite"
                    class="chakra-button css-15eygc7"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="27"
                      width="32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M0 8.9C0 3.872 4.332 0 9.417 0 12.038 0 14.345 1.252 16 2.871 17.655 1.251 19.962 0 22.583 0 27.668 0 32 3.872 32 8.9c0 3.445-1.442 6.474-3.441 9.035-1.995 2.556-4.605 4.725-7.104 6.488-.954.673-1.92 1.294-2.803 1.753-.83.43-1.784.824-2.652.824-.868 0-1.822-.394-2.652-.825-.884-.458-1.85-1.079-2.804-1.752-2.498-1.763-5.108-3.932-7.103-6.488C1.443 15.375 0 12.345 0 8.9Zm9.417-6.107c-3.802 0-6.674 2.847-6.674 6.107 0 2.614 1.09 5.047 2.845 7.297 1.76 2.254 4.127 4.243 6.519 5.93.904.638 1.757 1.181 2.486 1.56.783.406 1.236.52 1.407.52.17 0 .624-.114 1.407-.52.73-.379 1.582-.922 2.486-1.56 2.392-1.687 4.76-3.676 6.519-5.93 1.756-2.25 2.845-4.683 2.845-7.297 0-3.26-2.872-6.107-6.674-6.107-2.184 0-4.223 1.329-5.496 3.015-.26.343-.661.545-1.087.545-.426 0-.828-.202-1.087-.545C13.64 4.122 11.6 2.793 9.417 2.793Z"
                        fill="#979797"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
    <div
      class="css-1gvp5h0"
    >
      <a
        class="chakra-card css-1wt50ai"
        href="/detail/2"
      >
        <div
          class="chakra-stack css-m4lmkl"
        >
          <div
            class="css-1ql0rgo"
          >
            <img
              alt="Tasty Wooden Chair"
              class="chakra-image css-13rbvqk"
              src="https://loremflickr.com/640/480/fashion"
            />
          </div>
          <div
            class="chakra-card__body css-dspdqf"
          >
            <div
              class="chakra-stack css-1luc48q"
            >
              <div
                class="css-13rbvqk"
              >
                <h2
                  class="chakra-heading css-n93nk6"
                >
                  Tasty Wooden Chair
                </h2>
                <p
                  class="chakra-text css-1286dop"
                >
                  Ab repellat ullam autem. Esse facere facere debitis dicta quidem nihil. Saepe suscipit ex.
                </p>
                <span
                  class="chakra-badge css-1dm3gsd"
                >
                  <div
                    class="css-artlef"
                  >
                    available
                  </div>
                </span>
              </div>
              <div
                class="chakra-stack css-bot922"
              >
                <p
                  class="chakra-text css-17nnq38"
                >
                  $
                  430
                </p>
                <div
                  class="chakra-stack css-1y3f6ad"
                >
                  <button
                    aria-label="Button add to cart"
                    class="chakra-button css-p7mkev"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="27"
                      width="28"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M25.502 3.562H5.821L4.139.548A1.069 1.069 0 0 0 3.206 0H1.069a1.069 1.069 0 1 0 0 2.137h1.51l1.659 2.975 3.827 8.469.004.008.337.746-3.842 4.099a1.069 1.069 0 0 0 .658 1.793l3.503.4c4.38.5 8.804.5 13.186 0l3.502-.4a1.069 1.069 0 1 0-.242-2.124l-3.503.4c-4.22.483-8.48.483-12.7 0l-1.39-.159 2.82-3.008c.027-.03.052-.06.076-.09l1.072.14a21.81 21.81 0 0 0 4.534.114 13.237 13.237 0 0 0 9.071-4.658l.824-.976c.028-.032.053-.067.077-.103l1.535-2.34c1.088-1.657-.101-3.86-2.085-3.86Zm-14.949 9.539a.69.69 0 0 1-.538-.397l-.002-.004-3.164-7h18.653c.284 0 .454.314.298.551l-1.5 2.286-.782.926a11.1 11.1 0 0 1-7.607 3.906 19.674 19.674 0 0 1-4.089-.103l-1.269-.165Z"
                        fill="#979797"
                        fill-rule="evenodd"
                      />
                      <path
                        d="M7.48 22.442a2.137 2.137 0 1 0 0 4.275 2.137 2.137 0 0 0 0-4.275ZM21.017 24.58a2.137 2.137 0 1 1 4.275 0 2.137 2.137 0 0 1-4.275 0Z"
                        fill="#979797"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Button add to favorite"
                    class="chakra-button css-15eygc7"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="27"
                      width="32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M0 8.9C0 3.872 4.332 0 9.417 0 12.038 0 14.345 1.252 16 2.871 17.655 1.251 19.962 0 22.583 0 27.668 0 32 3.872 32 8.9c0 3.445-1.442 6.474-3.441 9.035-1.995 2.556-4.605 4.725-7.104 6.488-.954.673-1.92 1.294-2.803 1.753-.83.43-1.784.824-2.652.824-.868 0-1.822-.394-2.652-.825-.884-.458-1.85-1.079-2.804-1.752-2.498-1.763-5.108-3.932-7.103-6.488C1.443 15.375 0 12.345 0 8.9Zm9.417-6.107c-3.802 0-6.674 2.847-6.674 6.107 0 2.614 1.09 5.047 2.845 7.297 1.76 2.254 4.127 4.243 6.519 5.93.904.638 1.757 1.181 2.486 1.56.783.406 1.236.52 1.407.52.17 0 .624-.114 1.407-.52.73-.379 1.582-.922 2.486-1.56 2.392-1.687 4.76-3.676 6.519-5.93 1.756-2.25 2.845-4.683 2.845-7.297 0-3.26-2.872-6.107-6.674-6.107-2.184 0-4.223 1.329-5.496 3.015-.26.343-.661.545-1.087.545-.426 0-.828-.202-1.087-.545C13.64 4.122 11.6 2.793 9.417 2.793Z"
                        fill="#979797"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    class="chakra-stack css-17qdwnb"
  >
    <button
      class="chakra-button css-wjktab"
      disabled=""
      type="button"
    >
      Prev
    </button>
    <button
      class="chakra-button css-fv8jc9"
      type="button"
    >
      1
    </button>
    <button
      class="chakra-button css-wjktab"
      disabled=""
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Home page Match to snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-1ey9w6j"
  >
    <div
      class="chakra-input__group css-viwvyh"
      data-group="true"
    >
      <input
        class="chakra-input css-190aajt"
        placeholder="Search"
        value=""
      />
      <div
        class="css-1q3ec5g"
      >
        <svg
          fill="none"
          height="30"
          width="42"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M25.507 7.2H1.8a1.8 1.8 0 0 1 0-3.6h23.707a5.402 5.402 0 0 1 10.186 0H40.2a1.8 1.8 0 0 1 0 3.6h-4.507a5.402 5.402 0 0 1-10.186 0ZM40.2 26.4a1.8 1.8 0 0 0 0-3.6H16.493a5.402 5.402 0 0 0-10.186 0H1.8a1.8 1.8 0 0 0 0 3.6h4.507a5.402 5.402 0 0 0 10.186 0H40.2Z"
            fill="#6A6A6A"
          />
        </svg>
      </div>
    </div>
    <ul
      class="css-1c8ne0r"
      role="list"
    >
      <li
        class="css-0"
      >
        <a
          class="chakra-button css-16s5d4s"
          href="/?category=all"
        >
          All
        </a>
      </li>
      <li
        class="css-0"
      >
        <a
          class="chakra-button css-16s5d4s"
          href="/?category=fashion"
        >
          Fashion
        </a>
      </li>
      <li
        class="css-0"
      >
        <a
          class="chakra-button css-16s5d4s"
          href="/?category=electrics"
        >
          Electrics
        </a>
      </li>
      <li
        class="css-0"
      >
        <a
          class="chakra-button css-16s5d4s"
          href="/?category=books"
        >
          Books
        </a>
      </li>
    </ul>
  </div>
  <h2
    class="chakra-heading css-p4222r"
  >
    Trending Items
  </h2>
  <div
    class="css-3hictu"
  >
    <div
      class="css-1gvp5h0"
    >
      <a
        class="chakra-card css-1wt50ai"
        href="/detail/1"
      >
        <div
          class="chakra-stack css-m4lmkl"
        >
          <div
            class="css-1ql0rgo"
          >
            <img
              alt="Tasty Plastic Mouse"
              class="chakra-image css-13rbvqk"
              src="https://loremflickr.com/640/480/fashion"
            />
          </div>
          <div
            class="chakra-card__body css-dspdqf"
          >
            <div
              class="chakra-stack css-1luc48q"
            >
              <div
                class="css-13rbvqk"
              >
                <h2
                  class="chakra-heading css-n93nk6"
                >
                  Tasty Plastic Mouse
                </h2>
                <p
                  class="chakra-text css-1286dop"
                >
                  Quos tempora adipisci quidem ipsum.
                </p>
                <span
                  class="chakra-badge css-1dm3gsd"
                >
                  <div
                    class="css-artlef"
                  >
                    available
                  </div>
                </span>
              </div>
              <div
                class="chakra-stack css-bot922"
              >
                <p
                  class="chakra-text css-17nnq38"
                >
                  $
                  430
                </p>
                <div
                  class="chakra-stack css-1y3f6ad"
                >
                  <button
                    aria-label="Button add to cart"
                    class="chakra-button css-p7mkev"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="27"
                      width="28"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M25.502 3.562H5.821L4.139.548A1.069 1.069 0 0 0 3.206 0H1.069a1.069 1.069 0 1 0 0 2.137h1.51l1.659 2.975 3.827 8.469.004.008.337.746-3.842 4.099a1.069 1.069 0 0 0 .658 1.793l3.503.4c4.38.5 8.804.5 13.186 0l3.502-.4a1.069 1.069 0 1 0-.242-2.124l-3.503.4c-4.22.483-8.48.483-12.7 0l-1.39-.159 2.82-3.008c.027-.03.052-.06.076-.09l1.072.14a21.81 21.81 0 0 0 4.534.114 13.237 13.237 0 0 0 9.071-4.658l.824-.976c.028-.032.053-.067.077-.103l1.535-2.34c1.088-1.657-.101-3.86-2.085-3.86Zm-14.949 9.539a.69.69 0 0 1-.538-.397l-.002-.004-3.164-7h18.653c.284 0 .454.314.298.551l-1.5 2.286-.782.926a11.1 11.1 0 0 1-7.607 3.906 19.674 19.674 0 0 1-4.089-.103l-1.269-.165Z"
                        fill="#979797"
                        fill-rule="evenodd"
                      />
                      <path
                        d="M7.48 22.442a2.137 2.137 0 1 0 0 4.275 2.137 2.137 0 0 0 0-4.275ZM21.017 24.58a2.137 2.137 0 1 1 4.275 0 2.137 2.137 0 0 1-4.275 0Z"
                        fill="#979797"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Button add to favorite"
                    class="chakra-button css-15eygc7"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="27"
                      width="32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M0 8.9C0 3.872 4.332 0 9.417 0 12.038 0 14.345 1.252 16 2.871 17.655 1.251 19.962 0 22.583 0 27.668 0 32 3.872 32 8.9c0 3.445-1.442 6.474-3.441 9.035-1.995 2.556-4.605 4.725-7.104 6.488-.954.673-1.92 1.294-2.803 1.753-.83.43-1.784.824-2.652.824-.868 0-1.822-.394-2.652-.825-.884-.458-1.85-1.079-2.804-1.752-2.498-1.763-5.108-3.932-7.103-6.488C1.443 15.375 0 12.345 0 8.9Zm9.417-6.107c-3.802 0-6.674 2.847-6.674 6.107 0 2.614 1.09 5.047 2.845 7.297 1.76 2.254 4.127 4.243 6.519 5.93.904.638 1.757 1.181 2.486 1.56.783.406 1.236.52 1.407.52.17 0 .624-.114 1.407-.52.73-.379 1.582-.922 2.486-1.56 2.392-1.687 4.76-3.676 6.519-5.93 1.756-2.25 2.845-4.683 2.845-7.297 0-3.26-2.872-6.107-6.674-6.107-2.184 0-4.223 1.329-5.496 3.015-.26.343-.661.545-1.087.545-.426 0-.828-.202-1.087-.545C13.64 4.122 11.6 2.793 9.417 2.793Z"
                        fill="#979797"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
    <div
      class="css-1gvp5h0"
    >
      <a
        class="chakra-card css-1wt50ai"
        href="/detail/2"
      >
        <div
          class="chakra-stack css-m4lmkl"
        >
          <div
            class="css-1ql0rgo"
          >
            <img
              alt="Tasty Wooden Chair"
              class="chakra-image css-13rbvqk"
              src="https://loremflickr.com/640/480/fashion"
            />
          </div>
          <div
            class="chakra-card__body css-dspdqf"
          >
            <div
              class="chakra-stack css-1luc48q"
            >
              <div
                class="css-13rbvqk"
              >
                <h2
                  class="chakra-heading css-n93nk6"
                >
                  Tasty Wooden Chair
                </h2>
                <p
                  class="chakra-text css-1286dop"
                >
                  Ab repellat ullam autem. Esse facere facere debitis dicta quidem nihil. Saepe suscipit ex.
                </p>
                <span
                  class="chakra-badge css-1dm3gsd"
                >
                  <div
                    class="css-artlef"
                  >
                    available
                  </div>
                </span>
              </div>
              <div
                class="chakra-stack css-bot922"
              >
                <p
                  class="chakra-text css-17nnq38"
                >
                  $
                  430
                </p>
                <div
                  class="chakra-stack css-1y3f6ad"
                >
                  <button
                    aria-label="Button add to cart"
                    class="chakra-button css-p7mkev"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="27"
                      width="28"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M25.502 3.562H5.821L4.139.548A1.069 1.069 0 0 0 3.206 0H1.069a1.069 1.069 0 1 0 0 2.137h1.51l1.659 2.975 3.827 8.469.004.008.337.746-3.842 4.099a1.069 1.069 0 0 0 .658 1.793l3.503.4c4.38.5 8.804.5 13.186 0l3.502-.4a1.069 1.069 0 1 0-.242-2.124l-3.503.4c-4.22.483-8.48.483-12.7 0l-1.39-.159 2.82-3.008c.027-.03.052-.06.076-.09l1.072.14a21.81 21.81 0 0 0 4.534.114 13.237 13.237 0 0 0 9.071-4.658l.824-.976c.028-.032.053-.067.077-.103l1.535-2.34c1.088-1.657-.101-3.86-2.085-3.86Zm-14.949 9.539a.69.69 0 0 1-.538-.397l-.002-.004-3.164-7h18.653c.284 0 .454.314.298.551l-1.5 2.286-.782.926a11.1 11.1 0 0 1-7.607 3.906 19.674 19.674 0 0 1-4.089-.103l-1.269-.165Z"
                        fill="#979797"
                        fill-rule="evenodd"
                      />
                      <path
                        d="M7.48 22.442a2.137 2.137 0 1 0 0 4.275 2.137 2.137 0 0 0 0-4.275ZM21.017 24.58a2.137 2.137 0 1 1 4.275 0 2.137 2.137 0 0 1-4.275 0Z"
                        fill="#979797"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Button add to favorite"
                    class="chakra-button css-15eygc7"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="27"
                      width="32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M0 8.9C0 3.872 4.332 0 9.417 0 12.038 0 14.345 1.252 16 2.871 17.655 1.251 19.962 0 22.583 0 27.668 0 32 3.872 32 8.9c0 3.445-1.442 6.474-3.441 9.035-1.995 2.556-4.605 4.725-7.104 6.488-.954.673-1.92 1.294-2.803 1.753-.83.43-1.784.824-2.652.824-.868 0-1.822-.394-2.652-.825-.884-.458-1.85-1.079-2.804-1.752-2.498-1.763-5.108-3.932-7.103-6.488C1.443 15.375 0 12.345 0 8.9Zm9.417-6.107c-3.802 0-6.674 2.847-6.674 6.107 0 2.614 1.09 5.047 2.845 7.297 1.76 2.254 4.127 4.243 6.519 5.93.904.638 1.757 1.181 2.486 1.56.783.406 1.236.52 1.407.52.17 0 .624-.114 1.407-.52.73-.379 1.582-.922 2.486-1.56 2.392-1.687 4.76-3.676 6.519-5.93 1.756-2.25 2.845-4.683 2.845-7.297 0-3.26-2.872-6.107-6.674-6.107-2.184 0-4.223 1.329-5.496 3.015-.26.343-.661.545-1.087.545-.426 0-.828-.202-1.087-.545C13.64 4.122 11.6 2.793 9.417 2.793Z"
                        fill="#979797"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    class="chakra-stack css-17qdwnb"
  >
    <button
      class="chakra-button css-wjktab"
      disabled=""
      type="button"
    >
      Prev
    </button>
    <button
      class="chakra-button css-fv8jc9"
      type="button"
    >
      1
    </button>
    <button
      class="chakra-button css-wjktab"
      disabled=""
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
